### ISSUE TITLE
[Bug]: OpenGVLab/InternVL2-26B AsyncEngineDeadError: Task finished unexpectedly - value error in merge_vision_embeddings

### ISSUE THREAD

**Issue by marinapollo** [vllm-project/vllm#7272]
### Your current environment

```
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.3 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-1042-azure-x86_64-with-glibc2.35
Is CUDA available: N/A
CUDA runtime version: 12.2.140
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration:
GPU 0: NVIDIA A100 80GB PCIe
GPU 1: NVIDIA A100 80GB PCIe

Nvidia driver version: 535.54.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   48 bits physical, 48 bits virtual
Byte Order:                      Little Endian
CPU(s):                          48
On-line CPU(s) list:             0-47
Vendor ID:                       AuthenticAMD
Model name:                      AMD EPYC 7V13 64-Core Processor
CPU family:                      25
Model:                           1
Thread(s) per core:              1
Core(s) per socket:              48
Socket(s):                       1
Stepping:                        1
BogoMIPS:                        4890.89
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl tsc_reliable nonstop_tsc cpuid extd_apicid aperfmperf pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw topoext perfctr_core invpcid_single vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves clzero xsaveerptr rdpru arat umip vaes vpclmulqdq rdpid fsrm
Hypervisor vendor:               Microsoft
Virtualization type:             full
L1d cache:                       1.5 MiB (48 instances)
L1i cache:                       1.5 MiB (48 instances)
L2 cache:                        24 MiB (48 instances)
L3 cache:                        192 MiB (6 instances)
NUMA node(s):                    2
NUMA node0 CPU(s):               0-23
NUMA node1 CPU(s):               24-47
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Vulnerable
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] transformers==4.41.2
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] transformers              4.41.2                   pypi_0    pypi
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: N/A
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
GPU0    GPU1    NIC0    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      NV12    NODE    0-23    0               N/A
GPU1    NV12     X      SYS     24-47   1               N/A
NIC0    NODE    SYS      X

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_an0
```
```

### 🐛 Describe the bug

Tried to host and test `OpenGVLab/InternVL2-26B` model using vllm image [v0.5.4](https://hub.docker.com/layers/vllm/vllm-openai/v0.5.4/images/sha256-7ab0cf7b287876cec65752a1b7ac99790ecd2a609da80c4d1dd1fbeaf987abf6?context=explore) using 

- `--chat-template="InternVL2-template.jinja"  `
- `--load-format=safetensors `
- `--trust-remote-code `
- `--max_model_len 4096`

getting the error:

```

local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226] Exception in worker VllmWorkerProcess while processing method start_worker_execution_loop: Attempted to assign 7 x 256 = 1792 image tokens to 504 placeholders, Traceback (most recent call last):
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/vllm/executor/multiproc_worker_utils.py", line 223, in _run_worker_process
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     output = executor(*args, **kwargs)
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 116, in decorate_context
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     return func(*args, **kwargs)
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/worker_base.py", line 65, in start_worker_execution_loop
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     output = self.execute_model(execute_model_req=None)
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/worker_base.py", line 273, in execute_model
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     output = self.model_runner.execute_model(
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 116, in decorate_context
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     return func(*args, **kwargs)
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 1363, in execute_model
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     hidden_or_intermediate_states = model_executable(
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     return self._call_impl(*args, **kwargs)
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1562, in _call_impl
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     return forward_call(*args, **kwargs)
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/models/internvl.py", line 391, in forward
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     inputs_embeds = merge_vision_embeddings(input_ids, inputs_embeds,
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]   File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/models/utils.py", line 29, in merge_vision_embeddings
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]     raise ValueError(
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226] ValueError: Attempted to assign 7 x 256 = 1792 image tokens to 504 placeholders
local_llm_host_internVL2_instruct  | (VllmWorkerProcess pid=110) ERROR 08-07 14:32:00 multiproc_worker_utils.py:226]
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57] Engine background task failed
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57] Traceback (most recent call last):
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 47, in _log_task_completion
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     return_value = task.result()
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 642, in run_engine_loop
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     result = task.result()
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 585, in engine_step
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     request_outputs = await self.engine.step_async(virtual_engine)
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 254, in step_async
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     output = await self.model_executor.execute_model_async(
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/executor/distributed_gpu_executor.py", line 175, in execute_model_async
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     return await self._driver_execute_model_async(execute_model_req)
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/executor/multiproc_gpu_executor.py", line 224, in _driver_execute_model_async
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     return await self.driver_exec_model(execute_model_req)
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     result = self.fn(*self.args, **self.kwargs)
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/worker_base.py", line 273, in execute_model
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     output = self.model_runner.execute_model(
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 116, in decorate_context
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     return func(*args, **kwargs)
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 1363, in execute_model
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     hidden_or_intermediate_states = model_executable(
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     return self._call_impl(*args, **kwargs)
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1562, in _call_impl
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     return forward_call(*args, **kwargs)
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/models/internvl.py", line 391, in forward
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     inputs_embeds = merge_vision_embeddings(input_ids, inputs_embeds,
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]   File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/models/utils.py", line 29, in merge_vision_embeddings
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57]     raise ValueError(
local_llm_host_internVL2_instruct  | ERROR 08-07 14:32:00 async_llm_engine.py:57] ValueError: Attempted to assign 7 x 256 = 1792 image tokens to 504 placeholders
local_llm_host_internVL2_instruct  | INFO 08-07 14:32:00 async_llm_engine.py:181] Aborted request chat-f119ffe9e1e54cf3a8fe1d999db26ce9.
local_llm_host_internVL2_instruct  | Exception in callback _log_task_completion(error_callback=<bound method...7fc6e2de9a50>>)(<Task finishe...laceholders')>) at /usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py:37
local_llm_host_internVL2_instruct  | handle: <Handle _log_task_completion(error_callback=<bound method...7fc6e2de9a50>>)(<Task finishe...laceholders')>) at /usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py:37>
local_llm_host_internVL2_instruct  | Traceback (most recent call last):
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 47, in _log_task_completion
local_llm_host_internVL2_instruct  |     return_value = task.result()
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 642, in run_engine_loop
local_llm_host_internVL2_instruct  |     result = task.result()
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 585, in engine_step
local_llm_host_internVL2_instruct  |     request_outputs = await self.engine.step_async(virtual_engine)
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 254, in step_async
local_llm_host_internVL2_instruct  |     output = await self.model_executor.execute_model_async(
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/executor/distributed_gpu_executor.py", line 175, in execute_model_async
local_llm_host_internVL2_instruct  |     return await self._driver_execute_model_async(execute_model_req)
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/executor/multiproc_gpu_executor.py", line 224, in _driver_execute_model_async
local_llm_host_internVL2_instruct  |     return await self.driver_exec_model(execute_model_req)
local_llm_host_internVL2_instruct  |   File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
local_llm_host_internVL2_instruct  |     result = self.fn(*self.args, **self.kwargs)
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/worker_base.py", line 273, in execute_model
local_llm_host_internVL2_instruct  |     output = self.model_runner.execute_model(
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 116, in decorate_context
local_llm_host_internVL2_instruct  |     return func(*args, **kwargs)
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/worker/model_runner.py", line 1363, in execute_model
local_llm_host_internVL2_instruct  |     hidden_or_intermediate_states = model_executable(
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
local_llm_host_internVL2_instruct  |     return self._call_impl(*args, **kwargs)
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1562, in _call_impl
local_llm_host_internVL2_instruct  |     return forward_call(*args, **kwargs)
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/models/internvl.py", line 391, in forward
local_llm_host_internVL2_instruct  |     inputs_embeds = merge_vision_embeddings(input_ids, inputs_embeds,
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/models/utils.py", line 29, in merge_vision_embeddings
local_llm_host_internVL2_instruct  |     raise ValueError(
local_llm_host_internVL2_instruct  | ValueError: Attempted to assign 7 x 256 = 1792 image tokens to 504 placeholders
local_llm_host_internVL2_instruct  |
local_llm_host_internVL2_instruct  | The above exception was the direct cause of the following exception:
local_llm_host_internVL2_instruct  |
local_llm_host_internVL2_instruct  | Traceback (most recent call last):
local_llm_host_internVL2_instruct  |   File "/usr/lib/python3.10/asyncio/events.py", line 80, in _run
local_llm_host_internVL2_instruct  |     self._context.run(self._callback, *self._args)
local_llm_host_internVL2_instruct  |   File "/usr/local/lib/python3.10/dist-packages/vllm/engine/async_llm_engine.py", line 59, in _log_task_completion
local_llm_host_internVL2_instruct  |     raise AsyncEngineDeadError(
local_llm_host_internVL2_instruct  | vllm.engine.async_llm_engine.AsyncEngineDeadError: Task finished unexpectedly. This should never happen! Please open an issue on Github. See stack trace above for theactual cause.
```

1. `OpenGVLab/InternVL2-8B` works well however but only with the parameter `--max_model_len`. When using `OpenGVLab/InternVL2-8B` without `--max_model_len` similar error occurs (related issue found  [7068 OpenAI API for Phi-3-vision-128k-instruct](https://github.com/vllm-project/vllm/issues/7068) ) 

`OpenGVLab/InternVL2-26B model` is a different vision transformer model, may be this could be the problem? Related issue found [6322 VLLM 0.5.1 with LLaVA 1.6 exceptions](https://github.com/vllm-project/vllm/issues/6322)

**Comment by DarkLight1337** [comment]
Should be fixed by #7164

**Comment by Dineshkumar-Anandan-ZS0367** [comment]
Exceptions for 26B,
File "/usr/local/lib/python3.10/dist-packages/transformers/models/auto/auto_factory.py", line 732, in getitem
model_type = self._reverse_config_mapping[key.name]
KeyError: 'InternVLChatConfig'
Traceback (most recent call last):
File "/usr/local/lib/python3.10/dist-packages/vllm/entrypoints/openai/api_server.py", line 150, in build_async_engine_client
await async_engine_client.setup()
File "/usr/local/lib/python3.10/dist-packages/vllm/entrypoints/openai/rpc/client.py", line 35, in setup
await self.wait_for_server()
File "/usr/local/lib/python3.10/dist-packages/vllm/entrypoints/openai/rpc/client.py", line 136, in wait_for_server
await self._send_one_way_rpc_request(
File "/usr/local/lib/python3.10/dist-packages/vllm/entrypoints/openai/rpc/client.py", line 112, in _send_one_way_rpc_request
raise TimeoutError(f"server didn't reply within {timeout} ms")
TimeoutError: server didn't reply within 1000 ms
